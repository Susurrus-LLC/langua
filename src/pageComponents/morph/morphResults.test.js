import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import MorphResults from './morphResults'

describe('MorphResults', () => {
  const styles = {
    outText: 'outText',
    different: 'different',
    changed: 'changed',
    output: 'output'
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <MorphResults
        styles={styles}
        outputFormat='io'
        results={[
          { diff: true, input: 'lector', output: 'leitor' },
          { diff: false, input: 'doctor', output: 'doutor' },
          { diff: true, input: 'focus', output: 'focus' },
          { diff: false, input: 'jocus', output: 'jocus' }
        ]}
        showDiff={true}
        showChanges={true}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately with various props', () => {
    let tree = renderer
      .create(
        <MorphResults
          styles={styles}
          outputFormat='oo'
          results={[
            { diff: true, input: 'lector', output: 'leitor' },
            { diff: false, input: 'doctor', output: 'doutor' },
            { diff: true, input: 'focus', output: 'focus' },
            { diff: false, input: 'jocus', output: 'jocus' }
          ]}
          showDiff={true}
          showChanges={true}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderer
      .create(
        <MorphResults
          styles={styles}
          outputFormat='io'
          results={[
            { diff: true, input: 'lector', output: 'leitor' },
            { diff: false, input: 'doctor', output: 'doutor' },
            { diff: true, input: 'focus', output: 'focus' },
            { diff: false, input: 'jocus', output: 'jocus' }
          ]}
          showDiff={true}
          showChanges={true}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderer
      .create(
        <MorphResults
          styles={styles}
          outputFormat='oi'
          results={[
            { diff: true, input: 'lector', output: 'leitor' },
            { diff: false, input: 'doctor', output: 'doutor' },
            { diff: true, input: 'focus', output: 'focus' },
            { diff: false, input: 'jocus', output: 'jocus' }
          ]}
          showDiff={false}
          showChanges={false}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
