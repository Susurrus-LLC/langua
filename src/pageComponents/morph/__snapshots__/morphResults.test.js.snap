// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MorphResults renders accurately with various props 1`] = `
<div
  className="results"
>
  <div
    className="output"
  >
    <p
      className="outText different changed"
    >
      leitor
    </p>
    <p
      className="outText changed"
    >
      doutor
    </p>
    <p
      className="outText different"
    >
      focus
    </p>
    <p
      className="outText"
    >
      jocus
    </p>
  </div>
  <div
    className="stats"
  >
    <p
      className="statsText"
    >
      2 of 4 words changed (undefined unchanged); 2 of 4 words different from last time (2 the same)
    </p>
  </div>
</div>
`;

exports[`MorphResults renders accurately with various props 2`] = `
<div
  className="results"
>
  <div
    className="output"
  >
    <p
      className="outText different changed"
    >
      lector → leitor
    </p>
    <p
      className="outText changed"
    >
      doctor → doutor
    </p>
    <p
      className="outText different"
    >
      focus → focus
    </p>
    <p
      className="outText"
    >
      jocus → jocus
    </p>
  </div>
  <div
    className="stats"
  >
    <p
      className="statsText"
    >
      2 of 4 words changed (undefined unchanged); 2 of 4 words different from last time (2 the same)
    </p>
  </div>
</div>
`;

exports[`MorphResults renders accurately with various props 3`] = `
<div
  className="results"
>
  <div
    className="output"
  >
    <p
      className="outText"
    >
      leitor ← lector
    </p>
    <p
      className="outText"
    >
      doutor ← doctor
    </p>
    <p
      className="outText"
    >
      focus ← focus
    </p>
    <p
      className="outText"
    >
      jocus ← jocus
    </p>
  </div>
  <div
    className="stats"
  >
    <p
      className="statsText"
    >
      2 of 4 words changed (undefined unchanged); 2 of 4 words different from last time (2 the same)
    </p>
  </div>
</div>
`;
